File: BusConductor/BusConductor.cpp
/*
https://www.geeksforgeeks.org/problems/bus-conductor--170647/1
*/

class Solution {
  public:
    int findMoves(vector<int>& chairs, vector<int>& passengers) {
        sort(chairs.begin(), chairs.end());
        sort(passengers.begin(), passengers.end());
        int sz=chairs.size(), res=0;
        for(int i=0; i<sz; ++i){
            res+=abs(passengers[i]-chairs[i]);
        }
        return res;
    }
};

File: BusConductor/BusConductor.java
/*
https://www.geeksforgeeks.org/problems/bus-conductor--170647/1
*/

class Solution {
    public int findMoves(int[] chairs, int[] passengers) {
        Arrays.sort(chairs);
        Arrays.sort(passengers);
        int finalLength=chairs.length, resultantMoves=0;
        for(int i=0; i<finalLength; ++i){
            resultantMoves+=Math.abs(chairs[i]-passengers[i]);
        }
        return resultantMoves;
    }
}
